### PR TITLE
Reorganize the library as a monorepo

### PR DESCRIPTION
**Pull request by iandilling-adaptive**
This is pre-merge. I'll be starting the merge today, but more merging can be done later. We don't need to push this until everybody is ready. Here is a summary of the changes made and the current status of the repo:

* Code split into 4 separate workspaces:
   * A components package for custom components we build (public)
   * A theme package for our Material UI theme (public)
   * A docs package for our storybook (private)
   * A package to hold the custom SVGR template, used by our components and docs packages (private)
*  The package manager in use is now Yarn v3
   * I had to upgrade my local yarn version, following this guide: https://yarnpkg.com/getting-started/install
   * Migrated from NPM following this guide: https://yarnpkg.com/getting-started/migration
   * We're currently not using ZeroInstalls or PNP for backwards compatibility reasons, but we can discuss adding these features later.
* I've been having issues with the Docker image I'm using to test the Github actions workflows after migrating to Yarn, so I haven't been able to fully test them outside of dryrun mode. Yarn isn't installed in my local image, but it should be installed on GitHub's action runners. There may be some caveats with yarn running in Github Actions that I'm not aware of yet. I expect the Storybook GH pages and Linter will run fine, but the release workflow might not. I'll get to that when I work on creating an initial release.
* I will update the readme with instructions on how to build and develop in the new package structure once I complete the merge.

### CODE REVIEW

**Comment by iandilling-adaptive** [comment]
Update: I finished the first merge. The lint check is working now, too. Everything builds except the types in the `ui-kit-components` package. It depends on a typescript module augmentation of one of MUIs interfaces in the `ui-kit-theme` package, but it isn't being picked up. I've tried a few config changes here and there with no success, so I think I'll need to rewrite/reorganize the typescript configs.

File: package.json [hunk 1]
```diff
@@ -89,12 +49,13 @@
...
   "bugs": {
-    "url": "https://github.com/chuck-adaptive/ui-kit-2022/issues"
+    "url": "https://github.com/AdaptiveConsulting/ui-kit-2022/issues"
   },
   "readme": "ERROR: No README data found!",
-  "homepage": "https://github.com/chuck-adaptive/ui-kit-2022#readme",
-  "_id": "ui-kit-2022@0.0.0"
+  "homepage": "https://github.com/AdaptiveConsulting/ui-kit-2022#readme",
+  "_id": "ui-kit-2022@0.0.0",
+  "packageManager": "yarn@3.2.4"
```
Comment (tprince232) [inline package.json:60]: Interesting - was yarn version 3 required to get the mono-repo tooling working?

Do you have do a global install for yarn w/ this version?   (If so, might be nice to add that as a step in the `README.md`

File: packages/ui-kit-svgr-template/index.js [hunk 2]
```diff
@@ -32,4 +33,4 @@ const template = ({ imports, componentName, jsx, exports, interfaces }, { tpl })
   `;
 };
 
-export default template;
+module.exports = template;
```
Comment (tprince232) [inline packages/ui-kit-svgr-template/index.js:36]: Curious - what made you have to switch module import/export notation here?

Comment (tprince232) [reply packages/ui-kit-svgr-template/index.js:36]: Ah, I see this is a separate mini-package for the `svgr-template`, written in just `js`

File: package.json (hunk 1 above)
Comment (iandilling-adaptive) [reply package.json:60]: Yeah, the build commands require yarn2+. The newer yarn versions have better tooling when working with monorepos.  I'll add info about how to configure the local dev environment in the readme, as well as how to run the builds and start developing.

File: packages/ui-kit-svgr-template/index.js (hunk 2 above)
Comment (iandilling-adaptive) [reply packages/ui-kit-svgr-template/index.js:36]: I didn't think it was necessary to introduce another build step for this template, so I wrote it in plain js. I think our current tsconfig won't accept js files from other packages with ESM syntax. I can try to change it, as I'll be messing with the tsconfigs anyway to fix the type issue the components package currently has.

Comment (iandilling-adaptive) [reply packages/ui-kit-svgr-template/index.js:36]: Actually, scratch that, I remember now. The storybooks internal tsconfig, which is not straightforward to change, won't accept js files with ESM syntax, so I can only use CJS.

Comment (tprince232) [reply packages/ui-kit-svgr-template/index.js:36]: Got it - no need to change was mostly just curious.

**Comment by iandilling-adaptive** [comment]
I rebased onto the latest changes in main and updated the information in the README.

**Comment by tprince232** [comment]
Awesome no further comments from me!

**Review by wenfeng-xing — APPROVED** [review]

**Review by Honglin-Lu — APPROVED** [review]

**Pull request merged**